src/com/xilinx/rapidwright/timing/delayestimator/InterconnectInfo.java

```diff
@@ -0,0 +1,293 @@
+/*
+ *
+ * Copyright (c) 2021 Xilinx, Inc.
+ * All rights reserved.
+ *
+ * Author: Pongstorn Maidee, Xilinx Research Labs.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.xilinx.rapidwright.timing.delayestimator;
+
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.xilinx.rapidwright.timing.GroupDelayType;
+
+/**
+ * Encapsulate interconnect information for a device family.
+ * The info must be just accurate enough for estimating the min delay.
+ * This is for Ultrascale+.
+ */
+// TODO: Consider moving this to a file.  Leaving them in code is easier to be extended for future generations.
+// TODO: The key issue is it is not certain what features are enough to capture future generations.
+public class InterconnectInfo implements java.io.Serializable {
+
+    // TODO: Can we get this association from device model?
+    private Map<String,String> nodeToSitePin;
+
+
+    /**
+     * Define direction of routing resource in reference to INT tile coordinate.
+     */
+    // up is in increasing INT tile coordinate direction
+    public enum Direction {
```

Comment:
Can we standardize this enum somewhere more general?  Are there some enums that could be merged to serve the same purpose?